src/test/java/teammates/test/cases/automated/FeedbackSessionUnpublishedMailActionTest.java

```diff
@@ -0,0 +1,118 @@
+import teammates.common.util.EmailType;
+import teammates.common.util.EmailWrapper;
+import teammates.common.util.HttpRequestHelper;
+import teammates.common.util.TimeHelper;
+import teammates.common.util.Const.ParamsNames;
+import teammates.logic.automated.EmailAction;
+import teammates.logic.automated.FeedbackSessionUnpublishedMailAction;
+import teammates.logic.core.CoursesLogic;
+import teammates.logic.core.FeedbackSessionsLogic;
+import teammates.logic.core.InstructorsLogic;
+import teammates.logic.core.StudentsLogic;
+import teammates.test.util.Priority;
+
+@Priority(-1)
+public class FeedbackSessionUnpublishedMailActionTest extends BaseComponentUsingTaskQueueTestCase {
+    
+    private static final FeedbackSessionsLogic fsLogic = FeedbackSessionsLogic.inst();
+    private static final StudentsLogic studentsLogic = StudentsLogic.inst();
+    private static final InstructorsLogic instructorsLogic = InstructorsLogic.inst();
+    private static final DataBundle dataBundle = getTypicalDataBundle();
+    
+    @SuppressWarnings("serial")
+    public static class FeedbackSessionUnpublishedCallback extends BaseTaskQueueCallback {
+        
+        @Override
+        public int execute(URLFetchRequest request) {
+            
+            HashMap<String, String> paramMap = HttpRequestHelper.getParamMap(request);
+            
+            assertTrue(paramMap.containsKey(ParamsNames.EMAIL_TYPE));
+            EmailType typeOfMail = EmailType.valueOf(paramMap.get(ParamsNames.EMAIL_TYPE));
+            assertEquals(EmailType.FEEDBACK_UNPUBLISHED, typeOfMail);
+            
+            assertTrue(paramMap.containsKey(ParamsNames.EMAIL_FEEDBACK));
+            assertNotNull(paramMap.get(ParamsNames.EMAIL_FEEDBACK));
+            
+            assertTrue(paramMap.containsKey(ParamsNames.EMAIL_COURSE));
+            assertNotNull(paramMap.get(ParamsNames.EMAIL_COURSE));
+            
+            FeedbackSessionUnpublishedCallback.taskCount++;
+            return Const.StatusCodes.TASK_QUEUE_RESPONSE_OK;
+        }
+
+    }
+    
+    @BeforeClass
+    public static void classSetUp() throws Exception {
+        printTestClassHeader();
+        gaeSimulation.tearDown();
+        gaeSimulation.setupWithTaskQueueCallbackClass(FeedbackSessionUnpublishedCallback.class);
+        gaeSimulation.resetDatastore();
+        removeAndRestoreTypicalDataInDatastore();
+    }
+    
+    @AfterClass
+    public static void classTearDown() {
+        printTestClassFooter();
+    }
+    
+    @Test
+    public void testFeedbackSessionPublishedMailAction() throws Exception {
+
+        ______TS("Emails Test : activate all sessions with unpublished mails sent");
+        for (FeedbackSessionAttributes fs : dataBundle.feedbackSessions.values()) {
+            fs.setSentPublishedEmail(false);
+            fsLogic.updateFeedbackSession(fs);
+            assertFalse(fsLogic.getFeedbackSession(fs.getFeedbackSessionName(), fs.getCourseId()).isSentPublishedEmail());
```

Comment:
sorry, I just realized, we don't really need this here. we are taking care of unpublishing a session twice in `unpublishFeedbackSession` not in action.